File: Problema4/main.cs
/**
 * @file main.cs
 * @author Alan Davide Bovo (alandavide.bovo.stud@ispascalcomandini.it)
 * @date 2024-01-05
 * Compile with: mcs -out:main.exe main.cs
 */
using System;

public class MainClass {
    public static void Main(string[] args){
        Random rnd = new Random(0); // inizializzo il random con 0 per avere sempre gli stessi numeri
        Console.WriteLine("Welcome to the guessing game!"); // welcome message
        Console.WriteLine("You must guess 10 numbers that I'm thinking between 0 and 99");

        for(int i=0; i<10; i++){
            int guess, number = rnd.Next(100);                  // genero un numero random
            Console.Write($"Can you guess the {i+1} number? "); // chiedo all'utente di indovinare il numero
            // pls non rompere [il cazzo con i try catch IDEA DI GITHUB COPILOT]
            guess = int.Parse(Console.ReadLine());              // leggo il numero inserito dall'utente
            if(guess != number)                                 // controllo se il numero inserito è uguale a quello generato
                break;
            if(i == 9){                                         // controllo se è l'ultimo numero
                Console.WriteLine("Congratulations! You won the game!");
                return;
            }
        }
        Console.WriteLine("You lost the game!"); // se non è l'ultimo numero, l'utente ha perso
    }
}

File: Problema4/main.c
/**
 * @file main.c
 * @author Alan Davide Bovo (alandavide.bovo.stud@ispascalcomandini.it)
 * @date 2024-01-05
 * Compile with: gcc main.c -o main
 */
#include<stdlib.h>
#include<stdio.h>

int main(){
    srand(0); // inizializzazione del generatore di numeri casuali
    fputs("Welcome to the guessing game!\n", stdout); // stampa del messaggio di benvenuto
    fputs("You must guess 10 numbers that I'm thinking between 0 and 99\n", stdout);
    for(int i=0; i<10; i++){
        int guess, number = rand()%100;                 // generazione del numero casuale
        printf("Can you guess the %d number? ", i+1);   // stampa del messaggio di richiesta
        scanf("%d", &guess);                            // lettura del numero inserito dall'utente
        if(guess != number)                             // confronto tra il numero inserito e quello generato
            break;
        if(i == 9){                                     // se l'utente ha indovinato tutti i numeri
            fputs("Congratulations! You won the game!\n", stdout);
            return 0;
        }
    }
    fputs("You lost the game!\n", stdout); // stampa del messaggio di sconfitta
}